components/steering_wheel/src/IO_Digital.c

```diff
@@ -0,0 +1,64 @@
+/*
+ * IO_Digital.c
+ * Digital IO Module source code
+ */
+
+/******************************************************************************
+ *                             I N C L U D E S
+ ******************************************************************************/
+
+// Module Header
+#include "IO_Digital.h"
+
+// System includes
+#include <string.h>
+
+// Other includes
+#include "ModuleDesc.h"
+#include "Utility.h"
+#include "SystemConfig.h"
+
+/******************************************************************************
+ *                           P U B L I C  V A R S
+ ******************************************************************************/
+
+io_digital_S io_digital;
+
+/******************************************************************************
+ *                       P U B L I C  F U N C T I O N S
+ ******************************************************************************/
+
+/*
+ * @brief Initializes the Digital IO Init
+ */
+static void IO_Digital_init(void)
+{
+    memset(&io_digital, 0, sizeof(io_digital_S));
+}
+
+/**
+ * @brief Periodic function checking input states
+ */
+static void IO_Digital_10Hz_PRD(void)
+{
+    //Because of pull-up, signal is active low
+    io_digital.input.switch1 = (HAL_GPIO_ReadPin(SW1_GPIO_Port, SW1_Pin) == GPIO_PIN_RESET)
```

Comment:
while this does work, I was thinking of a more generic approach.

Make a struct type which has members for the GPIO port and pin, as well as a pointer to the destination (which you won't be able to do right now because you can't make a pointer to a member of a bitfield. However, I'm working on a FLAGS api which you can use here instead)

Then make a static array containing those structs, where each entry refers to one of the IO pins. You can make one of these arrays for each refresh rate we want to sample at (just one is fine for now). Then in each of the periodic functions, just iterate over the relevant array and do a read on that port/pin and populate the target variable.